### DISCUSSION TITLE
rust image lacks rustdoc binary

### DISCUSSION THREAD

**Discussion by hdhoang** [wolfi-dev/os#14495] [Q&A] [answered]
Hi

I tried using cgr rust image instead of docker library's one. However, it lacks the `rustdoc` binary. Doc tests and doc-site html generation need it.

```text
❯ podman run --rm -it --entrypoint '' cgr.dev/chainguard/rust@sha256:b5217e32dd39ad9bd2e2a183bde90d4dc94e20795f2c534c003c39390be60244 bash -c 'cargo init --lib && cargo test --doc'
     Created library package
   Compiling work v0.1.0 (/work)
    Finished test [unoptimized + debuginfo] target(s) in 0.20s
   Doc-tests work
error: doctest failed, to rerun pass `--doc`

Caused by:
  could not execute process `rustdoc --edition=2021 --crate-type lib --crate-name work --test src/lib.rs --test-run-directory /work -L dependency=/work/target/debug/deps -L dependency=/work/target/debug/deps --extern work=/work/target/debug/deps/libwork-f82762bb9bd855c0.rlib -C embed-bitcode=no --error-format human` (never executed)

Caused by:
  No such file or directory (os error 2)
```

According to `rustup`, that is normally provided by the main `rustc` component.
 I think something about the configure flags disabled it https://github.com/wolfi-dev/os/blob/c14a6028620d60c548b3ca0819bb2889cf3c6a4a/rust.yaml#L63-L65

Could you take a look? thanks, and also thanks for providing the awesome mindset on images.

**Comment by amouat** [comment]
Thanks, let me try to get an answer on this one. Did you open an issue on it?

**Reply by hdhoang** [reply]
sorry, I went dormant on this, and haven't verified anything further. please feel free to convert this thread to an issue

**Reply by amouat** [reply, marked as answer]
Looks like the rustdoc binary is present now.
